modules/meshcloud-metering-service-principal/module.tf

```diff
@@ -59,13 +59,17 @@ resource "azuread_service_principal" "meshcloud_metering" {
 // Create a password for the Enterprise application
 //---------------------------------------------------------------------------
 resource "time_rotating" "replicator_secret_rotation" {
+  count = var.create_password ? 1 : 0
```

Comment:
d: I liked the separation you did in replication SP with `auth.tf` for passwords, metering could also benefit from that imo.